File: LeetCode/1161.Maximum-Level-Sum-of-a-Binary-Tree.cpp
// 1161. Maximum Level Sum of a Binary Tree

class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);

        int ans = -1, mx_sum = INT_MIN, lvl = 0;
        while(!q.empty()) {
            int sz = q.size(), sum = 0;
            while(sz--) {
                TreeNode* curr = q.front();
                q.pop();
                sum += curr->val;
                if(curr->left) q.push(curr->left);
                if(curr->right) q.push(curr->right);
            }
            lvl++;

            if(sum > mx_sum) {
                mx_sum = sum;
                ans = lvl;
            }
        }

        return ans;
    }
};

File: LeetCode/1161-Maximum-Level-Sum-of-a Binary-Tree.java
// 1161. Maximum Level Sum of a Binary Tree

class Solution {
    public int maxLevelSum(TreeNode root) {
        int mxSum = Integer.MIN_VALUE;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int currLevel = 1, mxLevel = 1;
        while (!q.isEmpty()) {
            int sz = q.size();
            int currSum = 0;
            while (sz-- > 0) {
                TreeNode currNode = q.poll();
                currSum += currNode.val;
                if (currNode.left != null) q.add(currNode.left);
                if (currNode.right != null) q.add(currNode.right);
            }
            if (currSum > mxSum) {
                mxSum = currSum;
                mxLevel = currLevel;
            }
            currLevel++;
        }
        return mxLevel;
    }
}
